### DISCUSSION TITLE
Is not parallel loading metadata from disk when pool import？

### DISCUSSION THREAD

**Discussion by TimLand** [openzfs/zfs#13433] [Q&A]
<html>
<body>
<h3 dir="auto" style="box-sizing: border-box; margin-top: 0px !important; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">System information</h3><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"> zpool import takes about 14s </p>

Type | Version/Name
-- | --
Distribution Name | Ubuntu
Distribution Version | 16.04
Kernel Version | 4.4.20
Architecture | AMD64
OpenZFS Version | zfs-0.8.6

</body>
</html>

After immediately power off,  then import pool takes about 14s,  zvol sync attribute is set to “always”
Is there any way to improve the speed of the pool import？

ZFS dbdmsg:
In the 1651891685 ~ 1651891689 time range, it looks like loading data from disk is in order？
In the 1651891689 ~ 1651891698 time range, what did it do during this time take about 9 seconds ？

![7083a77498c618ab8179f050a0547b3](https://user-images.githubusercontent.com/9785555/167235990-8ba4387f-49b6-4af0-adff-cf0561eb1559.png)

**Comment by rincebrain** [comment]
You should probably try a version newer than December 2020 before lamenting performance issues, I think you'll find a number of things improved.

It's possible this won't be, of course, I don't recall all the changes that have happened since then, but I'd suggest trying before lamenting a functionality gap.

**Comment by amotin** [comment]
For example, there were my PRs #12789 and #13022 in just recent months.
